Client/Frontend/Home/TopSites/DataManagement/TopSitesDataUtility.swift

```diff
@@ -0,0 +1,26 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Storage
+
+protocol TopSitesDataUtilityInterface {
+    func removeSiteMatching(_ engine: OpenSearchEngine?, from sites: [Site]) -> [Site]
+}
+
+struct TopSitesDataUtility: TopSitesDataUtilityInterface {
+    func removeSiteMatching(_ engine: OpenSearchEngine?, from sites: [Site]) -> [Site] {
+        if let defaultSearchEngine = engine {
+            let filteredSites = sites.filter {
+                guard let secondLevelDomain = $0.secondLevelDomain else { return false }
+                let hasMatchedEngine = defaultSearchEngine.shortName.localizedCaseInsensitiveContains(secondLevelDomain)
+
+                return !hasMatchedEngine
+            }
+            return filteredSites
```

Comment:
bop the sucker in one go, without needing to even assign things! Arguably, you could even do it for the filter's return statement, but, with the length of the contains and the name of the variable, I think there's a good argument that the way the filter reads now is probably clearer than having it condensed. :)
```suggestion
            return sites.filter {
                guard let secondLevelDomain = $0.secondLevelDomain else { return false }
                let hasMatchedEngine = defaultSearchEngine.shortName.localizedCaseInsensitiveContains(secondLevelDomain)

                return !hasMatchedEngine
            }
```

Reply:
Yeah I felt it would read better (after replacing if let with guard let) this way over doing it concisely. But if you'd prefer it done in one shot I'm down!